### PR TITLE
Add MVP sample.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Haven't done MVP in a while, hope it's not bad haha :

### CODE REVIEW

File: samples/mvp/build.gradle [hunk 1]
```diff
@@ -0,0 +1,55 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+android {
+    buildToolsVersion versions.buildTools
```
Comment (vanniktech) [inline samples/mvp/build.gradle]: AGP Plugin takes care of that

File: samples/mvp/proguard-rules.pro [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Add project specific ProGuard rules here.
```
Comment (vanniktech) [inline samples/mvp/proguard-rules.pro]: Let's empty that file?

File: samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import com.nhaarman.mockito_kotlin.verify
+import io.reactivex.Observable
+import io.reactivex.observers.TestObserver
+import org.assertj.core.api.Assertions.*
+import org.junit.Test
+
+class SignInPresenterTest {
+
+    private val view = TestSignInView()
+    private val service = spy(TestSignInService())
```
Comment (vanniktech) [inline samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt]: can we get rid of the mockito disease here?

File: samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @Test
+    fun `sends request with valid credentials`() {
+        view.emailRelay.accept("test@email")
+        view.passwordRelay.accept("passw0rd")
+
+        view.signInSubmitActionsRelay.accept(Unit)
+
+        verify(service).signIn(Credentials(email = "test@email", password = "passw0rd"))
```
Comment (vanniktech) [inline samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt]: Can't we get the value from the relay?

File: samples/mvp/build.gradle (hunk 1 above)
Comment (artem-zinnatullin) [reply samples/mvp/build.gradle]: I like to be specific and avoid surprises with versioning :)

File: samples/mvp/proguard-rules.pro (hunk 2 above)
Comment (artem-zinnatullin) [reply samples/mvp/proguard-rules.pro]: yep 👍

File: samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt (hunk 4 above)
Comment (artem-zinnatullin) [reply samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt]: Damn it, Niklas!

I was like yeaah, but let me just copy-paste from mvvm one, noone will notice that it uses mockito

Comment (vanniktech) [reply samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt]: I do! :trollface:

Comment (artem-zinnatullin) [reply samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt]: Can't get value from relay here though, input != output

Comment (artem-zinnatullin) [reply samples/mvp/src/test/java/com/lyft/domic/samples/mvp/signin/SignInPresenterTest.kt]: See https://github.com/lyft/domic/pull/17/commits/d14383d8fdbaa08dad3d6df040c629adc140827c

**Comment by artem-zinnatullin** [comment]
Still used in Renderer tests

**Pull request merged**